### PR TITLE
Allow JSX Import Source to be overwritten

### PR DESCRIPTION
**Pull request by Interpause**
This pull request solves #46 by adding `jsxImportSource` to `PreactPluginOptions`. As a bonus, I changed `prepublishOnly` to `prepare`, which allows the package to be installed directly via Git and should not affect its behaviour when being published to npm.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
 		"dev": "vite demo",
 		"dev:build": "vite build demo",
 		"build": "rimraf dist && tsc && tsc -p tsconfig.cjs.json && node tools/postbuild.mjs",
-		"prepublishOnly": "npm run build"
```
Comment (JoviDeCroock) [inline package.json:19]: Let's leave this in

**Review by JoviDeCroock — APPROVED** [review]

**Review by kdubiel — APPROVED** [review]

**Comment by Interpause** [comment]
bump

**Review by marvinhagemeister — CHANGES_REQUESTED** [review]
This PR has the right spirit, but does a whole lot more than just allowing `jsxImportSource` to be overwritten.

**Comment by Interpause** [comment]
> This PR has the right spirit, but does a whole lot more than just allowing `jsxImportSource` to be overwritten.

Is this referring to the additional 3 commits caused by fast forwarding the pull request to main?

**Comment by rschristian** [comment]
You need to rebase your changes. The diff is quite mangled.

**Comment by Interpause** [comment]
Should be fixed now.

**Comment by JoviDeCroock** [comment]
@Interpause the comment to leave prePublish in still stands 😅

**Comment by rschristian** [comment]
@JoviDeCroock FWIW, `prepare` does run before a publish: https://docs.npmjs.com/cli/v8/using-npm/scripts#life-cycle-scripts

I would argue against `prepare` still though as there's no reason we need to run a build immediately after installing deps (which it does). That's just a nuisance, IMO.

File: package.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
...
+		"prepublishOnly": "npm run build",
+		"prepare": "npm run build"
```
Comment (marvinhagemeister) [inline package.json:21]: This causes the `build` command to be run whenever a user installs our vite preset. Is this change necessary to allow JSX Import Source to be overwritten?

Comment (rschristian) [reply package.json:21]: According to the docs, I think it's only when the preset is installed through git? Which, of course, would require `build` be ran.

Comment (marvinhagemeister) [reply package.json:21]: I think it's not just that:

> - Runs on local `npm install` without any arguments

In either way this change doesn't seem necessary to allow JSX import source to be overwritten.

Comment (rschristian) [reply package.json:21]: > I think it's not just that:

`microbundle` has a prepare script, for example, and it doesn't run on install (nor could it). "local" means this project, as in, if you were to clone this repo and run `npm install`.

Not the best wording from NPM there, to be fair.

> In either way this change doesn't seem necessary to allow JSX import source to be overwritten.

Yeah totally agree there, and I generally think it's a super annoying life cycle script anyways.

Comment (marvinhagemeister) [reply package.json:21]: I stand corrected!

Comment (Interpause) [reply package.json:21]: I am guessing to keep the pull request pure, you want to remove the changes to package.json?

Comment (rschristian) [reply package.json:21]: That, but also as it stands, you haven't provided an explanation for the change. Why are we adding a new script?

While we can guess about your intentions (as we've done above), for right now, it's a change without any rhyme nor reason.

Comment (Interpause) [reply package.json:21]: To be fair, the `prepare` script isn't needed if the package is installed from npm. However, as I needed to overwrite the import source for a project, and did not want to publish to npm, I added the `prepare` script so that installing via git would be possible.

**Comment by Interpause** [comment]
Say, its been a while. Actually, I think the changes made to allow configuring `jsxImportSource` is quite minor, such that it probably would not be too much trouble for a maintainer to just include it in a commit. If so, would it be fine to close this PR?

**Comment by o-alexandrov** [comment]
Could you please consider to merge the PR, as the changes had been cleaned up

File: src/index.ts [hunk 3]
```diff
@@ -130,7 +135,7 @@ function preactPlugin({
 							: "@babel/plugin-transform-react-jsx-development",
 						{
 							runtime: "automatic",
-							importSource: "preact",
+							importSource: jsxImportSource ?? "preact",
```
Comment (o-alexandrov) [inline src/index.ts:138]: nullish coalescing while having no benefit over `||` operator in this case, still shouldn't break anyone's environment, as it's supported in Node.js 14 (oldest version that hasn't reach EOL)

**Review by JoviDeCroock — APPROVED** [review]

**Review by marvinhagemeister — APPROVED** [review]
